app/services/katello/pulp3/repository/docker.rb

```diff
@@ -43,7 +43,7 @@ def recursive_add_api
         def remote_options
           options = {url: root.url, upstream_name: root.docker_upstream_name}
           if root.docker_tags_whitelist && root.docker_tags_whitelist.any?
-            options[:whitelist_tags] = root.docker_tags_whitelist.join(",")
```

Comment:
This might not be the complete fix. I created a docker repo, set the whitelist to looks like this:

```
irb(main):003:0> Katello::RootRepository.first.docker_tags_whitelist
=> ["latest", "uclibc"]
```

And then I updated the repo URL:

```
[vagrant@centos7-hammer-devel ~]$ hammer repository update --id=1 --url=http://registry-1.docker.io
Warning: An error occured while loading module hammer_cli_foreman_remote_execution.
Could not update the repository:
  Task 06caba61-de8f-4244-9d3e-99aad273800a: PulpDockerClient::ApiError: Error message: the server returns an error
  HTTP status code: 400
  Response headers: {"date"=>"Mon, 04 Nov 2019 20:59:07 GMT", "server"=>"gunicorn/19.9.0", "content-type"=>"application/json", "vary"=>"Accept,Cookie", "allow"=>"GET, POST, 
HEAD, OPTIONS", "x-frame-options"=>"SAMEORIGIN", "content-length"=>"69", "via"=>"1.1 centos7-katello-devel.strangeways.example.com", "connection"=>"close"}
  Response body: {"whitelist_tags":["Expected a list of items but got type \"str\"."]}
```

Reply:
how old was your pulp3 environment?

Reply:
fresh from today, set up per the wiki with your forklift PR

Reply:
Wait! Please stand by.

Reply:
Ok it works, I just forgot the part where I need to check out your PR ;)

Reply:
hah!